src/main/java/org/folio/consortia/service/impl/TenantServiceImpl.java

```diff
@@ -82,9 +100,13 @@ public Tenant save(UUID consortiumId, Tenant tenantDto) {
 
     checkTenantNotExistsAndConsortiumExistsOrThrow(consortiumId, tenantDto.getId());
     Tenant savedTenant = saveTenant(consortiumId, tenantDto);
-
+    User shadowAdminUser = userTenantService.prepareShadowUser(adminUserId, currentTenantContext.getTenantId(), currentTenantContext);
     runInFolioContext(createFolioExecutionContextForTenant(tenantDto.getId(), currentTenantContext, folioMetadata),
```

Comment:
I think we need to delete runInFolioContext here because it will run logic to create shadow admin and configuration asynchonusly, basically we need to do these actions synchronusly

Reply:
does it make sense to introduce new method executeAfterTransactionCommits when it's enough to replace 
`executeAfterTransactionCommits`
by
`try (var context = new FolioExecutionContextSetter`

Reply:
@NikitaSedyx :- I already tested your suggestion , there also we will have a different thread and it wont work in  one transaction.

Reply:
we will create tech debt story to investigate why transaction in this case not working properly, but for now lets use approach 
`try (var context = new FolioExecutionContextSetter`
because in this approach we are not creating new thread that that is suitable for us, because we want synchronus action here

Reply:
@SinghAdes after proposed change please add todo here regarding transaction behavior that you faced here